### PR TITLE
Add tree-building exercise

### PR DESCRIPTION
**Pull request by ErikSchierboom**

### CODE REVIEW

**Comment by robkeim** [comment]
@ErikSchierboom I'll let you fix the build for this one before I review it

**Comment by ErikSchierboom** [comment]
@robkeim Fixed!

File: exercises/tree-building/TreeBuildingTest.cs [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+        {
+            new TreeBuildingRecord { RecordId = 0, ParentId = 0 }
+        };
+
+        var tree = TreeBuilder.BuildTree(records);
+
+        AssertTreeIsLeaf(tree, id: 0);
+    }
+
+    [Test]
```
Comment (robkeim) [inline exercises/tree-building/TreeBuildingTest.cs]: This and all subsequent tests are missing the `Ignore` attribute

**Comment by robkeim** [comment]
Thanks @ErikSchierboom just one small change needed and I'll merge this PR.

File: exercises/tree-building/TreeBuildingTest.cs (hunk 1 above)
Comment (ErikSchierboom) [inline exercises/tree-building/TreeBuildingTest.cs]: Well, that is on purpose, as the refactoring exercises already have a working implementation. In fact, the refactoring should be done whilst keeping all tests passing.

Comment (robkeim) [inline exercises/tree-building/TreeBuildingTest.cs]: Totally right!

**Pull request merged**